libraries/Microsoft.Bot.Connector/Authentication/AdalAuthenticator.cs

```diff
@@ -120,11 +136,30 @@ private async Task<AuthenticationResult> AcquireTokenAsync(bool forceRefresh = f
                 // Always release the semaphore if we acquired it.
                 if (acquired)
                 {
-                    tokenRefreshSemaphore.Release();
+                    ReleaseSemaphore();
                 }
             }
         }
 
+        private void ReleaseSemaphore()
+        {
+            try
+            {
+                tokenRefreshSemaphore.Release();
+            }
+            catch (SemaphoreFullException)
+            {
+                // We should not be hitting this, but if we do hit it everything will keep working.
+                // Logging to have clear knowledge of whether this is happening.
+                logger?.LogWarning("Attempted to release a full semaphore.");
+            }
+            catch (Exception)
```

Comment:
Ha, I planned todo something else here but changed mymind. I'll remove this catch for general Exception and just let the exception flow through.